vlttng/venv.py

```diff
@@ -585,22 +585,27 @@ def _create(self):
         self._create_project_instructions()
 
         # build projects in this order
-        self._build_project('urcu')
-        self._build_project('popt')
-        self._build_lttng_ust()
-        self._build_project('libxml2')
-        self._build_project('lttng-tools')
-        self._build_project('lttng-modules')
-        self._build_project('glib')
-        self._build_project('elfutils')
-        self._build_project('babeltrace')
-        self._build_project('babeltrace2')
-        self._build_project('lttng-analyses')
-        self._build_project('tracecompass')
-        self._build_project('lttng-scope')
-
-        # create activate script
-        self._create_activate()
+        try:
```

Comment:
Would it work to just put `self._create_activate()` before the builds below?

Reply:
It should. I did that in my first test, but settled on try/finally because I didn't investigate to determine if any state used by `self._create_activate()` was updated or modified by the various preceding builds.

 I will rework the patch to move the call to `self._create_activate()` instead.